xcp_d/utils/bids.py

```diff
@@ -152,12 +152,20 @@ def collect_data(
     layout : pybids.layout.BIDSLayout
     subj_data : dict
     """
-    layout = BIDSLayout(
-        str(bids_dir),
-        validate=bids_validate,
-        derivatives=True,
-        config=["bids", "derivatives"],
-    )
+    if input_type != "hcp" or input_type != "dcan":
+        layout = BIDSLayout(
+            str(bids_dir),
+            validate=bids_validate,
+            derivatives=True,
+            config=["bids", "derivatives"],
+        )
+    else:
+        layout = BIDSLayout(
```

Comment:
Can you add a comment here why you set `derivatives=False` for dcan and hcp data?

Reply:
Your comment says that the files are already written into a derivatives directory, but that's true for fMRIPrep as well. The preprocessed data, whether it comes from HCP, DCAN, fMRIPrep, or Nibabies, is in a BIDS derivatives dataset.

Reply:
Yes!